proxy/backendtimeout_test.go

```diff
@@ -69,6 +73,13 @@ func TestFastService(t *testing.T) {
 }
 
 func TestBackendTimeoutInTheMiddleOfServiceResponse(t *testing.T) {
+	var logBuf bytes.Buffer
+	log.SetOutput(&logBuf)
+	defer log.SetOutput(os.Stderr)
```

Comment:
Lets use https://pkg.go.dev/testing#T.Cleanup

Reply:
I think it should be used only when the amount of code here will be much larger.
```go
testLog := NewTestLog()
defer testLog.Close()
```

This code is simple and straightforward. using `T.Cleanup` will make it look not so straightforward:
```go
testLog := NewTestLog()
t.Cleanup(func() {
    testLog.Close()
})
```